tests/unit/test_req_install.py

```diff
@@ -62,7 +62,7 @@ def test_install_req_from_string_without_comes_from(self):
 
         assert isinstance(install_req, InstallRequirement)
         assert install_req.link.url == wheel_url
-        assert install_req.req.url == wheel_url
+        assert install_req.req.url is None
```

Comment:
By altering these two unit tests, would we no longer be testing certain scenarios that we were testing before? I would make sure we're testing at least the same code paths as before, which may or may not require adding new unit test cases instead of simply modifying the existing ones.

Reply:
These are fairly recent tests from https://github.com/pypa/pip/pull/6336. I think assertions might be inspecting more of the internal state of the `InstallRequirement` than necessary. If I revert the changes to `src/pip/_internal/req/constructors.py` from https://github.com/pypa/pip/pull/6336/files#diff-adb783f5bb998606c3159f0fd47d6fdd, the tests fail as expected with the `AttributeError` #6336 was fixing:

```
>       if req.url and comes_from.link.netloc in domains_not_allowed:
E       AttributeError: 'NoneType' object has no attribute 'netloc'
```

Reply:
Okay, if both of those unit tests are still properly testing #6336, then my next question was going to be whether it makes sense to add a similar unit test of `install_req_from_string()` that tests your own changes to `install_req_from_string()`. This is what I was getting at in my previous comment when I was talking about possibly adding new tests. It would be good to have each unit test focused on testing one thing, with the name and docstring explicitly reflecting that, so it's not just e.g. accidentally testing certain things as a side effect.

Reply:
Tests added